fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -278,11 +278,7 @@ struct TransactionRateInfo {
 		smoothReleased.addDelta(numStartedAtPriority);
 	}
 
-	void disable() {
-		disabled = true;
-		rate = 0;
-		smoothRate.reset(0);
-	}
+	void disable() { disabled = true; }
```

Comment:
If master proxy is disconnected from RK for a long time, we should reduce the traffic and eventually to 0 to protect the cluster from being full.  

With that said, we can have an actor, monitoring the last time rate info is updated and reduce the RateLimit by half (configurable by knob) every KNOB->PROXY_RKUPDATE_TIMEOUT seconds. 

Also log the situation for easier debugging.

Reply:
An easier change would be, reduce the rate by half each time when leaseTimeout. WDYT?

Reply:
I was thinking that we would still effectively reduce the rate to 0 as before, the only thing that would change is that we don't return an error in the case where the lease is expired. That is the most targeted fix of the bug and is probably the safest choice for a quick patch.